classifier/bisac.py

```diff
@@ -0,0 +1,601 @@
+        m(Poetry, fiction, 'Novels in Verse'),
+        m(Poetry, fiction, 'Poetry'),
+        m(Reference_Study_Aids, nonfiction, "Language Arts"),
+        m(Romance, fiction, "Love & Romance"),
+        m(Science_Fiction, fiction, "Robots"),
+        m(Science_Fiction, fiction, "Time Travel"),
+        m(Social_Sciences, nonfiction, "Media Studies"),
+        m(Suspense_Thriller, fiction, 'Superheroes'),
+        m(Suspense_Thriller, fiction, 'Thrillers & Suspense'),
+
+        # Most of the subcategories of 'Science & Nature' go into Nature,
+        # but these go into Science.
+        m(Science, nonfiction, 'Science & Nature', 'Discoveries'),
+        m(Science, nonfiction, 'Science & Nature', 'Experiments & Projects'),
+        m(Science, nonfiction, 'Science & Nature', 'History of Science'),
+        m(Science, nonfiction, 'Science & Nature', 'Physics'),
+        m(Science, nonfiction, 'Science & Nature', 'Weights & Measures'),
+        m(Science, nonfiction, 'Science & Nature', 'General'),
+
+        # Any other subcategory of 'Science & Nature' goes under Nature
+        m(Nature, nonfiction, 'Science & Nature', something),
+
+        # Life Strategies is juvenile/YA-specific, and contains both
+        # fiction and nonfiction. It's called "Social Issues" for
+        # juvenile fiction/nonfiction, and "Social Topics" for YA
+        # nonfiction. "Social Themes" in YA fiction is _not_
+        # classified as Life Strategies.
+        m(Life_Strategies, fiction, "social issues"),
+        m(Life_Strategies, nonfiction, "social issues"),
+        m(Life_Strategies, nonfiction, "social topics"),
+    ]
+
+    @classmethod
+    def is_fiction(cls, identifier, name):
+        for ruleset in cls.FICTION:
+            fiction = ruleset.match(*name)
+            if fiction is cls.stop:
+                return None
+            if fiction is not None:
+                return fiction
+
+    @classmethod
+    def audience(cls, identifier, name):
+        for ruleset in cls.AUDIENCE:
+            audience = ruleset.match(*name)
+            if audience is cls.stop:
+                return None
+            if audience is not None:
+                return audience
+
+
+    @classmethod
+    def target_age(cls, identifier, name):
+        for ruleset in cls.TARGET_AGE:
+            target_age = ruleset.match(*name)
+            if target_age is cls.stop:
+                return None
+            if target_age is not None:
+                return target_age
+
+    @classmethod
+    def genre(cls, identifier, name, fiction, audience):
+        for ruleset in cls.GENRE:
+            genre = ruleset.match(*name)
+            if genre is cls.stop:
+                return None
+            if genre is not None:
+                return genre
+
+    @classmethod
+    def scrub_name(cls, name):
+        """Split the name into a list of lowercase keywords."""
+        return [x.strip().lower() for x in name.split('/')]
```

Comment:
I didn't see anything that uses the NAMES mapping. Should this check if the name is in fact an identifier, and replace it with the name? Or remove the identifier if the name starts with an identifier?

Reply:
NAMES is used in the test. There is a lot more cleanup work that can happen here, like using different separators or removing stuff that was copied-and-pasted from the BISAC web site, but I haven't seen these specific problems yet.

Reply:
I thought Biblioboard used the identifiers rather than the names?

Feedbooks also uses the identifiers, but they prepend 'FB' to them, so they would need a special scrub_name.

Reply:
The Feedbooks code would go into scrub_identifier. With them and Biblioboard we should be able to create a Subject that uses both the identifier and the name. 

I can write some code that replaces our built-in name for the identifier (in case the distributor gives the wrong name) but I think that code would happen prior to scrub_name being called, maybe instead of scrub_name being called.